go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -99,6 +99,45 @@ func (qre *QueryExecutor) shouldConsolidate() bool {
 	return cm == tabletenv.Enable || (cm == tabletenv.NotOnPrimary && qre.tabletType != topodatapb.TabletType_PRIMARY)
 }
 
+func (qre *QueryExecutor) shouldConsolidateStmt() bool {
+	// If for whatever reason we don't have a stmt to evaluate,
+	// default to vttablet-level settings.
+	if qre.plan.Statement == nil {
+		return qre.shouldConsolidate()
+	}
+
+	// If the stmt isn't commented in, default to vttablet-level settings.
+	c, isC := qre.plan.Statement.(sqlparser.Commented)
+	if !isC {
+		return qre.shouldConsolidate()
+	}
+
+	p := c.GetParsedComments()
+
+	// If we can't parse the comments, default to vttablet-level settings.
+	if p == nil {
+		return qre.shouldConsolidate()
+	}
+
+	d := p.Directives()
+
+	// If none of the comments are directives, default to vttablet-level
+	// settings.
+	if p.Directives() == nil {
+		return qre.shouldConsolidate()
+	}
+
```

Comment:
this part will not work, from vtgate we do not pass down comments directlive, the only way to receive it is through ExecuteOptions and the problem with that is some default needs to be set on it (eg. false is default), which will defeat the purpose that whether client wants to disable the consolidation or the client have not set the value.

I think we cannot achieve all the cases, I would suggest that choose a path based on the requirement 
i.e. 
client can only enable query consilidation otherwise default of vttablet will be considered.

Reply:
Hey @harshit-gangal understood about not being able to rely on comments directives being passed down from VTG to VTT. 

Regarding `ExecuteOptions`, could we do what appears to be done with `workload`, where the default value is `UNSPECIFIED`? In that case we would rely on VTT-level settings.

Reply:
Using enum it can be done.

Reply:
I updated the PR to parse the comment directive and update the VTGate session options, which are then passed down to VTTablet. Let me know if you think the approach looks right. I will add tests and docs once I know I'm on the right track.